test/general/buffer_wrapper.pass.cpp

```diff
@@ -48,7 +50,8 @@ main()
     std::size_t size = 1000;
     sycl::buffer<std::uint32_t> buf{size};
     test_buffer_wrapper test{};
-    auto data_ptr = sycl::host_accessor<std::uint32_t, 1, sycl::access_mode::read>(buf).get_pointer();
+    sycl::host_accessor buf_accessor(buf, sycl::read_only);

```

Comment:
In this line we use constructor as described at https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#_interface_for_buffer_host_accessors :
```
  /* Available only when: (Dimensions > 0) */
  template <typename AllocatorT, typename TagT>
  host_accessor(buffer<DataT, Dimensions, AllocatorT>& bufferRef, TagT tag,
                const property_list& propList = {});
```

In the SYCL source code from DPCPP compiler it constructor from the file: https://github.com/intel/llvm/blob/sycl/sycl/include/sycl/accessor.hpp, line 3019 :
```
  template <typename T = DataT, int Dims = Dimensions, typename AllocatorT,
            typename = detail::enable_if_t<IsSameAsBuffer<T, Dims>::value>>
  host_accessor(
      buffer<T, Dims, AllocatorT> &BufferRef, mode_tag_t<AccessMode>,
      const property_list &PropertyList = {},
      const detail::code_location CodeLoc = detail::code_location::current())
      : host_accessor(BufferRef, PropertyList, CodeLoc) {}
```